modules/network/src/main/java/org/apache/ignite/internal/network/recovery/RecoveryDescriptor.java

```diff
@@ -124,4 +128,45 @@ public long onReceive() {
     public String toString() {
         return S.toString(RecoveryDescriptor.class, this);
     }
+
+    private final AtomicReference<Channel> ctxHolder = new AtomicReference<>();
+
+    /**
+     * Release this descriptor.
+     *
+     * @param ctx Channel handler context.
+     */
+    public void release(ChannelHandlerContext ctx) {
+        ctxHolder.compareAndSet(ctx.channel(), null);
```

Comment:
Should we assert the result, just in case?

Reply:
Not really, we don't care if release was called more than once